### PR TITLE
[Localization]Localized move-trigger text fix

### PR DESCRIPTION
**Pull request by mercurius-00**
## What are the changes?

Localized move-trigger text like:
- _"was damaged by the recoil!"_
- _" cut its own HP to power up its move!"_
- _"took {{targetName}}\ninto the sky!"_
- _"incinerated\n{{targetName}}'s {{itemName}}!"_
- _......_

## Why am I doing these changes?

It should be localized

## What did change?

- Extract the text from [data/move.ts](https://github.com/mercurius-00/pokerogue/blob/main/src/data/move.ts) to [locale/[language]/move-trigger.ts](https://github.com/mercurius-00/pokerogue/blob/main/src/locales/en/move-trigger.ts)
- Change some English text to match the original game.
    | original                                      | changed                                         |
    | ------------------------------------------- | ----------------------------------------------- |
    | " is hit with recoil!"                      | " was damaged by the recoil!"                   |
    | "took in sunlight!"                         | "absorbed light!"                               |
    | "dug a hole!"                               | "burrowed its way under the ground!"            |
    | "lowered\nits head!"                        | "tucked in its head!"                           |
    | "is glowing!"                               | "became cloaked in a harsh light!"              |
    | "is charging its power!"                    | "is absorbing power!"                           |
    | " is trying\nto take its foe down with it!" | " is hoping to take its attacker down with it!" |
- Fixed type name localization

### Screenshots/Videos

<details>

<summary>✅En (Fully tested)</summary>

1. STRUGGLE
image
2. STEEL_BEAM
image
3. COVET
image
4. INCINERATE
image
5. KNOCK_OFF
image
6. ELECTRO_SHOT
image
7. FUTURE_SIGHT
image
8. DOOM_DESIRE
image
9. BELLY_DRUM
image
10. PSYCH_UP
image
12. HEART_SWAP
image
13. FICKLE_BEAM
image
14. MAGNITUDE
image
15. PRESENT
image
16. JUMP_KICK
image
17. LOCK_ON
image
18. WHIRLWIND
image
19. CAMOUFLAGE
image
20. SOAK
image
21. MIMIC
image
22. SKETCH
image
23. WORRY_SEED
image
24. ROLE_PLAY
image
25. DOODLE
image
26. ENTRAINMENT
image
27. SKILL_SWAP
image
28. TRANSFORM
image
29. PAY_DAY
image
30. DESTINY_BOND
image
31. POLTERGEIST
image
32. RAZOR_WIND
33. FLY
image
34. SOLAR_BEAM
image
35. DIG
image
36. SKULL_BASH
image
37. SKY_ATTACK
image
38. HEAL_BELL
image
39. DIVE
image
40. AROMATHERAPY
image
41. BOUNCE
image
42. SHADOW_FORCE
image
43. SKY_DROP
image
44. FREEZE_SHOCK
image
45. ICE_BURN
image
46. PHANTOM_FORCE
image
47. GEOMANCY
image
48. SOLAR_BLADE
image
49. BURN_UP
image
50. BEAK_BLAST
image
51. METEOR_BEAM
image
52. DOUBLE_SHOCK
image
53. FORESTS_CURSE
image
54. TRICK_OR_TREAT
image

</details>

<details>
<summary>⬜Es</summary>

Missing

</details>

<details>
<summary>✅It</summary>

SKY_DROP
image
KNOCK_OFF
image
TRANSFORM
image
SOAK
image
TRICK_OR_TREAT
image
POLTERGEIST
image
COVET
BELLY_DRUM
image
ROLE_PLAY
image

</details>

<details>
<summary>✅Fr</summary>

SKY_DROP
image
KNOCK_OFF
image
TRANSFORM
image
SOAK
image
TRICK_OR_TREAT
image
POLTERGEIST
image
COVET
image
BELLY_DRUM
image
ROLE_PLAY
image

</details>

<details>
<summary>✅De</summary>

SKY_DROP
image
KNOCK_OFF
image
TRANSFORM
image
SOAK
image
TRICK_OR_TREAT
image
POLTERGEIST

COVET

BELLY_DRUM

ROLE_PLAY
image

</details>

<details>
<summary>✅Pt_BR</summary>

SKY_DROP
image
KNOCK_OFF
image
TRANSFORM
image
SOAK
image
TRICK_OR_TREAT
image
POLTERGEIST
image
COVET
image
BELLY_DRUM
image
ROLE_PLAY
image

</details>

<details>
<summary>✅Zh-CN</summary>

SKY_DROP
image
KNOCK_OFF
image
TRANSFORM
image
SOAK
image
TRICK_OR_TREAT
image
POLTERGEIST
image
COVET
image
BELLY_DRUM
image
ROLE_PLAY
image

</details>

<details>
<summary>✅Zh-TW</summary>

SKY_DROP
image
KNOCK_OFF
image
TRANSFORM
image
SOAK
image
TRICK_OR_TREAT
image
POLTERGEIST
image
COVET
image
BELLY_DRUM
image
ROLE_PLAY
image

</details>

<details>
<summary>✅Ko</summary>

SKY_DROP
image
KNOCK_OFF
image
TRANSFORM
image
SOAK
image
TRICK_OR_TREAT

POLTERGEIST
image
COVET
image
BELLY_DRUM
image
ROLE_PLAY
image

</details>

## How to test the changes?

1. set overrides.ts
`export const MOVESET_OVERRIDE: Array<Moves> = [Moves.SKY_DROP, Moves.BEAK_BLAST, Moves.MIND_BLOWN, Moves.STEEL_BEAM];`
2. run `npm run  start:dev`
3. enter a battle and use moves

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/ko/move-trigger.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  "magnitudeMessage": "매그니튜드{{magnitude}}!",
+  "tookAimAtTarget": "{{pokemonName}}[[는]] 목표를\n{{targetName}}[[로]] 결정했다!",
+  "transformedIntoType": "{{pokemonName}}[[는]]\n{{typeName}}타입이 됐다!",
+  "copiedMove": "{{pokemonName}}[[는]]\n{{moveName}}[[를]] 복사했다!",
+  "sketchedMove": "{{pokemonName}}[[는]]\n{{moveName}}[[를]] 스케치했다!",
+  "acquiredAbility": "{{pokemonName}}[[는]]\n{{abilityName}}[[가]] 되었다!",
+  "copiedTargetAbility": "{{pokemonName}}[[는]] {{targetName}}의\n{{abilityName}}[[를]] 복사했다!",
+  "transformedIntoTarget": "{{pokemonName}}[[는]]\n{{targetName}}[[로]] 변신했다!",
+  "tryingToTakeFoeDown": "{{pokemonName}}[[는]] 상대를\n길동무로 삼으려 하고 있다!",
+  "addType": "{{typeName}} was added to\n{{pokemonName}}!",
```
Comment (EnochG1) [inline src/locales/ko/move-trigger.ts:52]:
```suggestion
  "addType": "{{pokemonName}}에게\n{{typeName}}타입이 추가되었다!",
```

**Review by EnochG1 — COMMENTED** [review]
Korean translation
++ Could you add Korean screenshots of any message including type?

File: src/locales/de/move-trigger.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  "magnitudeMessage": "Intensität {{magnitude}}!",
+  "tookAimAtTarget": "{{pokemonName}} zielt auf {{targetName}}!",
+  "transformedIntoType": "{{pokemonName}} nimmt den Typ {{typeName}} an!",
+  "copiedMove": "{{pokemonName}} kopiert {{moveName}}!",
+  "sketchedMove": "{{pokemonName}} ahmt die Attacke {{moveName}} nach!",
+  "acquiredAbility": "The {{pokemonName}} nimmt die Fähigkeit {{abilityName}} an!",
+  "copiedTargetAbility": "{{pokemonName}} kopiert{{abilityName}} von  {{targetName}}!",
+  "transformedIntoTarget": "{{pokemonName}} verwandelt sich in {{targetName}}!",
+  "tryingToTakeFoeDown": "{{pokemonName}} versucht, den Angreifer mit sich zu nehmen!",
+  "addType": "{{typeName}} was added to\n{{pokemonName}}!",
```
Comment (CodeTappert) [inline src/locales/de/move-trigger.ts:52]:
```suggestion
  "addType": "{{pokemonName}} nimmt zusätzlich den Typ {{typeName}} an!",
```

File: src/locales/pt_BR/move-trigger.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  "magnitudeMessage": "Magnitude {{magnitude}}!",
+  "tookAimAtTarget": "{{pokemonName}} mirou\nem {{targetName}}!",
+  "transformedIntoType": "{{pokemonName}} se transformou\nem tipo {{typeName}}!",
+  "copiedMove": "{{pokemonName}} copiou\no movimento {{moveName}}!",
+  "sketchedMove": "{{pokemonName}} desenhou\no movimento {{moveName}}!",
+  "acquiredAbility": "{{pokemonName}} adquiriu\na habilidade {{abilityName}}!",
+  "copiedTargetAbility": "{{pokemonName}} copiou a habilidade\nde {{targetName}}!",
+  "transformedIntoTarget": "{{pokemonName}} se transformou\nem um(a) {{targetName}}!",
+  "tryingToTakeFoeDown": "{{pokemonName}} está tentando derrubar o atacante com ele!",
+  "addType": "{{typeName}} was added to\n{{pokemonName}}!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/move-trigger.ts:52]:
```suggestion
  "addType": "{{pokemonName}} recebeu\no tipo {{typeName}}!",
```

File: src/locales/fr/move-trigger.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  "magnitudeMessage": "Ampleur {{magnitude}} !",
+  "tookAimAtTarget": "{{pokemonName}} vise\n{{targetName}} !",
+  "transformedIntoType": "{{pokemonName}} prend\nle type {{typeName}} !",
+  "copiedMove": "{{pokemonName}} copie\nla capacité {{moveName}} !",
+  "sketchedMove": "{{pokemonName}} utilise Gribouille\npour copier {{moveName}} !",
+  "acquiredAbility": "Le talent de {{pokemonName}}\ndevient {{abilityName}} !",
+  "copiedTargetAbility": "{{pokemonName}} copie le talent\n{{abilityName}} de {{targetName}} !",
+  "transformedIntoTarget": "{{pokemonName}} prend\nl’apparence de {{targetName}} !",
+  "tryingToTakeFoeDown": "{{pokemonName}} veut entrainer\nson assaillant dans sa chute !",
+  "addType": "{{typeName}} was added to\n{{pokemonName}}!",
```
Comment (Adri1) [inline src/locales/fr/move-trigger.ts:52]:
```suggestion
  "addType": "{{pokemonName}} gagne\nle type {{typeName}}.",
```

**Review by Adri1 — COMMENTED** [review]
Added French for move-trigger line 52

**Comment by NicusPulcis** [comment]
@mercurius-00 the PR i've sent is now complete. everything has been localized for italian in said PR.

unfortunately, the only issue remains the one with moves that should include gender identification for items, as italian (and french and spanish as well i think) works in a weird way with that. the translations for those moves (incinerate, theft and knock off) aren't the exact ones from the games, since we don't have any way of determinating the item's gender, but they work regardless. i've asked for help on this topic already, but i'm pretty sure this translation is perfect as it is for now.

File: src/locales/ko/move-trigger.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  "sprangUp": "{{pokemonName}}[[는]]\n높이 뛰어올랐다!",
+  "choseDoomDesireAsDestiny": "{{pokemonName}}[[는]]\n파멸의소원을 미래에 맡겼다!",
+  "vanishedInstantly": "{{pokemonName}}의 모습이\n일순간에 사라졌다!",
+  "tookTargetIntoSky": "{{pokemonName}}[[는]] {{targetName}}[[를]]\n상공으로 데려갔다!",
+  "becameCloakedInFreezingLight": "{{pokemonName}}[[는]]\n차가운 빛에 둘러싸였다!",
+  "becameCloakedInFreezingAir": "{{pokemonName}}[[는]]\n차디찬 공기에 둘러싸였다!",
+  "isChargingPower": "{{pokemonName}}[[는]]\n파워를 모으고 있다!",
+  "burnedItselfOut": "{{pokemonName}}의 불꽃은 다 타 버렸다!",
+  "startedHeatingUpBeak": "{{pokemonName}}[[는]]\n부리를 가열하기 시작했다!",
+  "isOverflowingWithSpacePower": "{{pokemonName}}에게서\n우주의 힘이 넘쳐난다!",
```
Comment (EnochG1) [inline src/locales/ko/move-trigger.ts:35]:
```suggestion
  "absorbedElectricity": "{{pokemonName}}는(은)\n전기를 흡수했다!",
  "switchedStatChanges": "{{pokemonName}}[[는]] 상대와 자신의\n능력 변화를 바꿨다!",
  "goingAllOutForAttack": "{{pokemonName}}[[는]]\n전력을 다하기 시작했다!",
  "regainedHealth": "{{pokemonName}}[[는]]\n기력을 회복했다!",
  "keptGoingAndCrashed": "{{pokemonName}}[[는]]\n의욕이 넘쳐서 땅에 부딪쳤다!",
  "fled": "{{pokemonName}}[[는]]\N도망쳤다!",
  "cannotBeSwitchedOut": "{{pokemonName}}[[를]]\n돌아오게 할 수 없습니다!",
  "swappedAbilitiesWithTarget": "{{pokemonName}}[[는]]\n서로의 특성을 교체했다!",
  "coinsScatteredEverywhere": "돈이 주위에 흩어졌다!",
  "attackedByItem": "{{itemName}}[[가]]\n{{pokemonName}}에게 덤벼들었다!",
  "whippedUpAWhirlwind": "{{pokemonName}}의 주위에서\n공기가 소용돌이친다!",
  "flewUpHigh": "{{pokemonName}}는(은)\n하늘 높이 날아올랐다!",
  "tookInSunlight": "{{pokemonName}}는(은)\n빛을 흡수했다!",
  "dugAHole": "{{pokemonName}}는(은)\n땅으로 파고들었다!",
  "loweredItsHead": "{{pokemonName}}는(은)\n목을 움츠렸다!",
  "isGlowing": "{{pokemonName}}를(을)\n강렬한 빛이 감쌌다!",
  "bellChimed": "방울소리가 울려 퍼졌다!",
  "foresawAnAttack": "{{pokemonName}}는(은)\n미래의 공격을 예지했다!",
  "hidUnderwater": "{{pokemonName}}는(은)\n물속에 몸을 숨겼다!",
  "soothingAromaWaftedThroughArea": "기분 좋은 향기가 퍼졌다!",
  "sprangUp": "{{pokemonName}}는(은)\n높이 뛰어올랐다!",
  "choseDoomDesireAsDestiny": "{{pokemonName}}는(은)\n파멸의소원을 미래에 맡겼다!",
  "vanishedInstantly": "{{pokemonName}}의 모습이\n일순간에 사라졌다!",
  "tookTargetIntoSky": "{{pokemonName}}는(은) {{targetName}}를(을)\n상공으로 데려갔다!",
  "becameCloakedInFreezingLight": "{{pokemonName}}는(은)\n차가운 빛에 둘러싸였다!",
  "becameCloakedInFreezingAir": "{{pokemonName}}는(은)\n차디찬 공기에 둘러싸였다!",
  "isChargingPower": "{{pokemonName}}는(은)\n파워를 모으고 있다!",
  "burnedItselfOut": "{{pokemonName}}의 불꽃은 다 타 버렸다!",
  "startedHeatingUpBeak": "{{pokemonName}}는(은)\n부리를 가열하기 시작했다!",
  "isOverflowingWithSpacePower": "{{pokemonName}}에게서\n우주의 힘이 넘쳐난다!",
```

Temporarily change postposition.

**Review by EnochG1 — COMMENTED** [review]
@mercurius-00 
Korean has an issue of {USER} and {TARGET}.
The Pokémon's name determines the postposition(like [[는]], [[를]], changes based on the preceding letter), and some part doesn't work correctly because i18next.t is acted on first and the Pokémon name is replaced later.
So I've temporarily changed it to show both postpositions.

absorbedElectricity - ElectroShotChargeAttr(ChargeAttr) - Moves.ELECTRO_SHOT
flewUpHigh - ChargeAttr - Moves.FLY
tookInSunlight - SunlightChargeAttr(ChargeAttr) - Moves.SOLAR_BEAM
dugAHole - ChargeAttr - Moves.DIG
loweredItsHead - ChargeAttr - Moves.SKULL_BASH
isGlowing - ChargeAttr - Moves.SKY_ATTACK
foresawAnAttack - DelayedAttackAttr - Moves.FUTURE_SIGHT
hidUnderwater - ChargeAttr - Moves.DIVE
sprangUp - ChargeAttr - Moves.BOUNCE
choseDoomDesireAsDestiny - DelayedAttackAttr - Moves.DOOM_DESIRE
tookTargetIntoSky - ChargeAttr - Moves.SKY_DROP
becameCloakedInFreezingLight - ChargeAttr - Moves.FREEZE_SHOCK
becameCloakedInFreezingAir - ChargeAttr - Moves.ICE_BURN
isChargingPower - ChargeAttr - Moves.GEOMANCY
isGlowing - SunlightChargeAttr(ChargeAttr) - Moves.SOLAR_BLADE
startedHeatingUpBeak - ChargeAttr - Moves.BEAK_BLAST

**Review by EnochG1 — APPROVED** [review]

**Comment by NicusPulcis** [comment]
i've pretty much encountered the same issue as @EnochG1 btw, and i solved it just like she did: added all of the possible articles (il/lo/la) and that'll work as long as we don't have a way of determinating genders

**Comment by CodeTappert** [comment]
PLease fix the conflicts

**Comment by mercurius-00** [comment]
> PLease fix the conflicts

fixed

**Comment by NicusPulcis** [comment]
@mercurius-00 i just noticed i made a typo:

Line 6 of the italian localization, shouldn't be "aassorbe" but "assorbe", is it still possible for you to edit it?

File: src/locales/it/move-trigger.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
+import { SimpleTranslationEntries } from "#app/interfaces/locales";
+
+export const moveTriggers: SimpleTranslationEntries = {
+  "hitWithRecoil" : "{{pokemonName}} ha subito il contraccolpo!",
+  "cutHpPowerUpMove": "{{pokemonName}} riduce i suoi PS per potenziare la sua mossa!",
+  "absorbedElectricity": "{{pokemonName}} aassorbe elettricità!",
```
Comment (mercurius-00) [inline src/locales/it/move-trigger.ts:6]:
```suggestion
  "absorbedElectricity": "{{pokemonName}} assorbe elettricità!",
```

**Review by mercurius-00 — COMMENTED** [review]
fix typo

**Comment by mercurius-00** [comment]
> @mercurius-00 i just noticed i made a typo:
> 
> Line 6 of the italian localization, shouldn't be "aassorbe" but "assorbe", is it still possible for you to edit it?

Of course, you can also edit it; just start a review like I did.

**Comment by CodeTappert** [comment]
Please add the missing screenshot in english. And if you have time all screenshots for all languages (but that is optional)

**Comment by mercurius-00** [comment]
> @mercurius-00 i just noticed i made a typo:
> 
> Line 6 of the italian localization, shouldn't be "aassorbe" but "assorbe", is it still possible for you to edit it?



> Please add the missing screenshot in english. And if you have time all screenshots for all languages (but that is optional)

I believe I have provided all the related English screenshots

**Comment by Tempo-anon** [comment]
The beta branch got messed up. Do you mind reopening this PR with the merge conflicts fixed, or creating a new PR with your current changes?

**Pull request closed without merging**